app/src/main/java/de/tum/in/tumcampusapp/api/tumonline/model/Error.kt

```diff
@@ -1,7 +1,30 @@
 package de.tum.`in`.tumcampusapp.api.tumonline.model
 
-import org.simpleframework.xml.Element
-import org.simpleframework.xml.Root
+import com.tickaroo.tikxml.annotation.PropertyElement
+import com.tickaroo.tikxml.annotation.Xml
+import de.tum.`in`.tumcampusapp.api.tumonline.exception.*
+import java.io.IOException
 
-@Root(name = "error", strict = false)
-data class Error(@field:Element(name = "message", required = false) var message: String = "")
+@Xml(name = "error")
+data class Error(@PropertyElement var message: String = "") {
+
+    val exception: IOException
+        get() = ERROR_MESSAGE_TO_EXCEPTION.entries
+                .filter { message.contains(it.key) }
+                .map { it.value }
+                .firstOrNull() ?: UnknownErrorException()
+
+    companion object {
+
+        private val ERROR_MESSAGE_TO_EXCEPTION = mapOf(
```

Comment:
Same as above.
Linear searching in a map is rather slow, so use a List instead

Reply:
Done.